src/compiler/parser.ts

```diff
@@ -436,6 +436,13 @@ module ts {
         // attached to the EOF token.
         let parseErrorBeforeNextFinishedNode: boolean = false;
 
+        export const enum StatementFlags {
+            None = 0,
+            Statement = 1,
+            ModuleElement = 2,
+            StatementOrModuleElement = Statement | ModuleElement
+        }

```

Comment:
This is not really what I expected. I feel like all Statements are ModuleElements, but not all ModuleElements are statements. So I expected something more along the lines of
```ts
None = 0,
Statement = 1,
ModuleElement = 3,
```